hnn_core/dipole.py

```diff
@@ -48,6 +50,12 @@ def simulate_dipole(net, n_trials=None):
     if n_trials < 1:
         raise ValueError("Invalid number of simulations: %d" % n_trials)
 
+    if record_vsoma is not None and isinstance(record_vsoma, bool):
+        net.params['record_vsoma'] = record_vsoma
+    else:
+        raise TypeError("record_vsoma must be bool, got %s"
+                        % type(record_vsoma).__name__)
+
```

Comment:
Hi @ntolley @jasmainak! I had a followup on this PR. Enforcing a bool here causes an issue with how the HNN GUI displays parameter values. Normally, the user enters 0 or 1, which gets saved to the params dict. Here, it changes that value and the GUI shows "False", which is not suitable for a text box. I could change the GUI to automatically reconvert to 0 or 1 pretty easily, but this seems to be the first parameter hnn-core enforces. Is that the convention you are aiming for?

BTW, I realize a better GUI solution would be checkboxes but am kind of hoping hnn-core will allow 0 or 1 and I can leave the code as-is.

Reply:
I'd defer to @jasmainak's opinion for this one, but I but I believe this has come up in the past and the decision was to avoids `int`s for boolean values.

Reply:
yeah, we're aiming to have this to be boolean in `hnn-core`. Do you mind type-casting it in HNN for now? We can convert to checkboxes in the future!

Reply:
sure, thanks for the clarification